diff --git a/source/session/plugins/package.d b/source/session/plugins/package.d
index 965c64f..84bc5ed 100644
--- a/source/session/plugins/package.d
+++ b/source/session/plugins/package.d
@@ -34,13 +34,11 @@ Plugin[] insPlugins;
     Initializes Lua support
 */
 void insLuaInit() {
-    // LuaSupport support = loadLua();
-
-    // if (support == LuaSupport.noLibrary || support == LuaSupport.badLibrary) {
-    //     couldLoadLua = false;
-    //     insLogWarn("Could not load Lua support...");
-    // } else insLogInfo("Lua support initialized.");
-    insLogInfo("Lua support initialized. (Statically linked for now)");
+    LuaSupport support = loadLua("libluajit-5.1.so.2");
+    if (support == LuaSupport.noLibrary || support == LuaSupport.badLibrary) {
+        couldLoadLua = false;
+        insLogWarn("Could not load Lua support...");
+    } else insLogInfo("Lua support initialized.");
 
     // Create Lua state
     state = new LuaState(luaL_newstate());
@@ -55,6 +53,7 @@ void insLuaInit() {
 
 void insLuaUnload() {
     lua_close(state.handle());
+    unloadLua();
     destroy(state);
 }
 
