diff --git a/dub.sdl b/dub.sdl
index 2ccf86e..e3fba5f 100644
--- a/dub.sdl
+++ b/dub.sdl
@@ -17,7 +17,6 @@ dependency "mir-linux-kernel" version="~>1.0.1"
 targetPath "out/"
 workingDirectory "out/"
 dflags "-mscrtlib=msvcrt" platform="windows-ldc"
-dflags "-link-defaultlib-shared=false" platform="linux-ldc"
 lflags "-rpath=$$ORIGIN" platform="linux"
 versions "GL_32" "USE_SDL2" "SDL_2020" "UseUIScaling"
 stringImportPaths "res/" "./"
@@ -135,4 +134,4 @@ configuration "update-version" {
 configuration "meta" {
 	targetType "none"
 	preGenerateCommands "rc.exe /v build-aux\\windows\\inochi-creator.rc" platform="windows"
-}
+}
\ No newline at end of file
